test/release/examples/primers/arrays.chpl

```diff
@@ -52,7 +54,7 @@ writeln("After assigning two elements, A is: ", A);
 
 //
 // Arrays can also be accessed using ranges to refer to subsets of
-// array elements, or sub-arrays, using a technique called slicing:
+// array elements, or *sub-arrays*, using a technique called slicing:
```

Comment:
I consider this is a general technical term (and Google seems to support this), not particularly specific to Chapel (and much more common than sub-domain, just because arrays are so much more common than domains), and think we should feel free to use it / define it more than we do.

`git grep`ping the sources, I get 9 hits outside this file (and 10 more in third-party) and 43 hits for 'subarray' (with 3 more in third-party), though admittedly most of these are in code and comments rather than documentation (I think `localSlice()` is the only case that uses the term in its chpldocumentation, and I very well may have introduced that one as well).

That said, I'm not insistent on introducing it in this primer if you're opposed to it.